src/Facebook/Url/FacebookUrlDetectionHandler.php

```diff
@@ -95,7 +95,7 @@ protected function protocolWithActiveSsl($protocol)
     protected function getHostName()
     {
         // Check for proxy first
-        if ($host = $this->getHeader('X_FORWARDED_HOST')) {
+        if ($this->isValidForwardedHost() && $host = $this->getHeader('X_FORWARDED_HOST')) {
```

Comment:
I'd write `if ($header = $this->getHeader('X_FORWARDED_HOST') && $this->isValidForwardedHost($header))`. It will also simplify the function